src/main/java/net/fabricmc/loader/game/MinecraftGameProvider.java

```diff
@@ -158,6 +159,31 @@ public void acceptArguments(String... argStrs) {
 		FabricLauncherBase.processArgumentMap(arguments, envType);
 	}
 
+	@Override
+	public String[] getLaunchArguments(boolean sanitize) {
+		if (arguments != null) {
+			List<String> list = new ArrayList<>(Arrays.asList(arguments.toArray()));
+			if (sanitize) {
+				int remove = 0;
+				Iterator<String> iterator = list.iterator();
+				while (iterator.hasNext()) {
+					String next = iterator.next();
+					if ("--accessToken".equals(next)) {
+						remove = 2;
```

Comment:
This assumes the access token is always passed in the form `--accessToken token` (2 arguments), however, the game will also accept it being passed as `--accessToken=token` (one argument).
Now, I don't know of any launcher currently doing that, but you should still handle that case.

Reply:
wait nvm.. didn't catch the Arguments#toArray() call handling that for you..